RCCL: Add Memory Manager to the documentation (#7741)
ROCm/rocm-systems: projects/rccl/docs/api-reference/library-specification.rst

.. doxygenfunction:: ncclCommUserRank

.. _communicator-suspend-resume:

Communicator suspend and resume
-------------------------------

These functions release and reacquire the resources held by a communicator
that is temporarily idle, and report per-communicator memory statistics. They
are useful when an application wants to free GPU memory held by an inactive

> communicator and later resume collective operations on the same communicator.

Reviewer: ```suggestion
communicator, and then later resume collective operations on the same communicator.
```

Releasing the physical backing of a suspended communicator requires virtual
memory management (VMM) support enabled through ``NCCL_CUMEM_ENABLE``. Without

> it, ``ncclCommSuspend`` and ``ncclCommResume`` succeed but do not release GPU

Reviewer: ```suggestion
it, ``ncclCommSuspend`` and ``ncclCommResume`` succeed but don't release GPU
```

memory. See :ref:`suspend-resume` for the full list of prerequisites.

> To suspend or resume several communicators atomically, wrap the calls in

Reviewer 2: The note about ncclGroupStart/ncclGroupEnd for atomic suspend/resume appears both in library-specification.rst and rccl-usage-tips.rst. Fine to have it in both places, but the spec section is very brief — consider adding  a sentence about what "atomically" means here (i.e., the suspend/resume operations are submitted as a group  but are they truly atomic from the GPU's perspective, or just submitted together?).

Author: Thanks! Fixed
